README.md

```diff
@@ -10,21 +10,65 @@ npm install @bundled-es-modules/pdfjs-dist
 
 ## Use
 
-```html
-<script type="module">
-  import { pdfjs } from '@bundled-es-modules/pdfjs-dist/index.js';
-  pdfjs.GlobalWorkerOptions.workerSrc = '@bundled-es-modules/pdfjs-dist/pdf.worker.js';
-  var url = 'basicapi.pdf';
-  var loadingTask = pdfjs.getDocument(url);
-  //...
-</script>
+```js
+import pdfjs from "@bundled-es-modules/pdfjs-dist/build/pdf";
+import viewer from "@bundled-es-modules/pdfjs-dist/web/pdf_viewer";
+
+pdfjs.GlobalWorkerOptions.workerSrc =
+  "@bundled-es-modules/pdfjs-dist/pdf.worker.js";
+
+var url = "basicapi.pdf";
+var loadingTask = pdfjs.getDocument(url);
+//...
 ```
 
-## Overview of this mirror repo
+**Important**: Unlike in the original build, the `pdfjs.GlobalWorkerOptions.workerSrc` is mandatory.
+
+# Overview
+
+### Background
+
+PDFJS is a library by Mozilla, created to facillitate client-side rendering of PDF documents.
+
+Currently, the primary build provided for this project is [pdfjs/pdfjs-dist].
+
+Unfortunately, it's currently only distributed as UMD-style module. This is problematic for authors of modern applications, as it prevents leveraging the benefits of ES modules.
+
+Therefore, this mirror repository seeks to provide an alternative bundle of the library, allowing consumption from within modern web modules.
+
+_Support for es modules in the main project has been requested [here (#10317)](https://github.com/mozilla/pdf.js/issues/10317)_
+
+### Structure
+
+```
+├── build/
+│   ├── pdf.js                    - display layer
+│   └── pdf.worker.js             - core layer
+└── web/
+    ├── pdf_viewer.js             - top-level viewer import, containing all necessary setup
+    ├── module.js                 - pdf-viewer module, transformed from UMD to ESM
+    └── util.js                   - utility module for setup
+
+```
+
+### Rollup configuration
```

Comment:
My preference would be to leave this commentary inside the rollup config itself.  I like the idea of outlining intent, but most consumers probably won't care so long as the output works as intended.